89920 Add upgrade CLI for diff providers (#4508)
mesosphere/dcos-docs-site: pages/dkp/konvoy/2.3/dkp-upgrade/upgrade-konvoy/essential/index.md

<p class="message--warning"><strong>IMPORTANT:</strong>If you have more than one essential cluster, ensure your <code>dkp</code> configuration references the management cluster where you want to run the upgrade by setting the <code>KUBECONFIG</code> environment variable, or using the <code>--kubeconfig</code> flag, <a href="https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/">in accordance with Kubernetes conventions</a>.

> Upgrade the core addons in a cluster using the 'dkp upgrade addons' command specifying the cluster infrastructure (choose [aws, azure, vsphere, gcp, eks, preprovisioned]) and the name of the cluster.  In the examples below, replace infrastructure provider in command at the location with corresponding provider name as shown in brackets above.

Reviewer: ```suggestion
Upgrade the core addons in a cluster using the `dkp upgrade addons` command specifying the cluster infrastructure (choose [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`]) and the name of the cluster. In the examples below, replace infrastructure provider in command at the location with corresponding provider name as shown in brackets above.
```

    - If an AMI was specified when initially creating a cluster for AWS, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-ami/">Konvoy Image Builder</a>.
    - If an Azure Machine Image was specified for Azure, you must build a new one with <a href="/dkp/konvoy/2.3/image-builder/create-azure-image/">Konvoy Image Builder</a>.

> 1.  Upgrade the Kubernetes version of the control plane.  As shown in update addons section above, replace infrastructure provider name in the command. [aws, azure, vsphere, gcp, eks, preprovisioned].

Reviewer: ```suggestion
1.  Upgrade the Kubernetes version of the control plane. As shown in update addons section above, replace infrastructure provider name in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

     ✓ Updating the control plane
    ```

> 1.  Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool.  Also, replace the name of the infrastructure provider in the command [aws, azure, vsphere, gcp, eks, preprovisioned].

Reviewer: ```suggestion
1.  Upgrade the Kubernetes version of each of your node pools. Replace `my-nodepool` with the name of the node pool. Also, replace the name of the infrastructure provider in the command [`aws`, `azure`, `vsphere`, `gcp`, `eks`, `preprovisioned`].
```

[envariables]: ../../../choose-infrastructure/azure/quick-start-azure#configure-azure-prerequisites
[envariables2]: ../../../choose-infrastructure/aws/quick-start-aws#configure-aws-prerequisites
[envariables3]: ../../../choose-infrastructure/vsphere/new#create-a-new-vsphere-kubernetes-cluster

> [envariables4]: ../../../choose-infrastructure/eks/quick-start-eks#configure-eks-prerequisites

Reviewer: I don't know why - but github UI freaked out and had to select all three of them - only the last line needs a change.

> [envariables4]: ../../../choose-infrastructure/eks/quick-start-eks#configure-eks-prerequisites

Author: Thank you sir!
